.github/workflows/publish.yml

```diff
@@ -0,0 +1,149 @@
+
+    - name: Checkout tags
+      uses: openttd/actions/checkout@v1
+      with:
+        with-tags: true
+
+    - name: Set variables
+      id: vars
+      run: |
+        set -e
+
+        FULL_VERSION=$(./findversion.sh)
+        RELEASE_DATE=$(TZ='UTC' date +"%Y-%m-%d %H:%M UTC")
+        VERSION=$(echo "${FULL_VERSION}" | cut -f 1 -d$'\t')
+
+        # If this secret is not set, we are doing a dryrun. This means that
+        # people who fork this repository will not try to publish this to AWS
+        # (as that will most likely fail anyway).
+        if [ -z "${{ secrets.CDN_S3_BUCKET }}" ]; then
+          DRY_RUN="true"
+        else
+          DRY_RUN="false"
+        fi
+
+        REPO_NAME=$(echo ${{ github.repository }} | cut -d/ -f2 | tr [A-Z] [a-z])
+        # If we run on "schedule", we are producting a nightly. Otherwise it
+        # is a tag, so a release.
+        if [ "${{ github.event_name }}" = "schedule" ]; then
+          FOLDER="${REPO_NAME}-nightlies"
+
+          # Download the latest version we published; if we are different, it
+          # is safe to assume we are newer.
+          LATEST_VERSION=$(curl --fail -s https://cdn.openttd.org/${FOLDER}/latest.yaml | grep version | cut -d: -f2 | cut -b 2-)
+
+          if [ "${LATEST_VERSION}" = "${VERSION}" ]; then
+            echo "Run on schedule; going to skip this run, as we already build this version"
+            SKIP="true"
+          elif [ "${DRY_RUN}" = "true" ]; then
+            # Also skip everything if we are triggered frm the schedule and we
+            # are a fork. That would waste a lot of CPU cycles for no good reason.
+            echo "Run on schedule; going to skip this run, as this is a fork"
+            SKIP="true"
+          else
+            SKIP="false"
+          fi
+        else
+          FOLDER="${REPO_NAME}-releases"
+          SKIP="false"
+        fi
+
+        echo "::set-output name=release-date::${RELEASE_DATE}"
+        echo "::set-output name=version::${VERSION}"
+        echo "::set-output name=folder::${FOLDER}"
+        echo "::set-output name=dry-run::${DRY_RUN}"
+        echo "::set-output name=skip::${SKIP}"
+
+        echo "Release-date: ${RELEASE_DATE}"
+        echo "Version: ${VERSION}"
+        echo "Folder: ${FOLDER}"
+        echo "Dry-run: ${DRY_RUN}"
+        echo "Skip: ${SKIP}"
+
+    - if: steps.vars.outputs.skip == 'false'
+      name: Install dependencies
+      run: |
+        set -e
+
+        sudo apt update
+        sudo apt install -y catcodec --no-install-recommends
+
+    - if: steps.vars.outputs.skip == 'false'
+      name: Build
+      run: |
+        set -e
+
+        make maintainer-clean
+        make bundle_zip bundle_xsrc
+
+        # Move bundles in their own folder
+        mkdir bundles
+        mv opensfx-${{ steps.vars.outputs.version }}-source.tar.xz bundles/
+        mv opensfx-${{ steps.vars.outputs.version }}.zip bundles/opensfx-${{ steps.vars.outputs.version }}-all.zip
```

Comment:
Maybe it would be better to fix Makefile to add "-all"
I guess it should be Makefile:352